pkg/utils/k8sutil.go

```diff
@@ -769,16 +475,29 @@ func splitHandler(handler string) (string, string, error) {
 	return str[0], str[1], nil
 }
 
-func ensureFuncConfigMap(client kubernetes.Interface, funcObj *spec.Function, or []metav1.OwnerReference) error {
+func getFileName(file, handler, runtime string) (string, error) {
+	// DEPRECATED: If the filename is empty, assume that
+	// the destination file will be <handler>.<ext>
+	if file == "" {
+		modName, _, err := splitHandler(handler)
+		if err != nil {
+			return "", err
+		}
+		return kubelessRuntime.GetFunctionFileName(modName, runtime), nil
+	}
+	return file, nil
+}
+
+// EnsureFuncConfigMap creates/updates a config map with a function specification
+func EnsureFuncConfigMap(client kubernetes.Interface, funcObj *spec.Function, or []metav1.OwnerReference) error {
 	configMapData := map[string]string{}
 	var err error
 	if funcObj.Spec.Handler != "" {
-		modName, _, err := splitHandler(funcObj.Spec.Handler)
+		fileName, err := getFileName(funcObj.Spec.File, funcObj.Spec.Handler, funcObj.Spec.Runtime)
 		if err != nil {
 			return err
 		}
-		fileName, depName := GetFunctionFileNames(funcObj.Spec.Runtime, modName)
-
+		depName, _ := kubelessRuntime.GetRuntimeDepName(funcObj.Spec.Runtime)
```

Comment:
Why is it safe to ignore errors here?

Reply:
for unsupported runtimes it won't return a `depName` but we can still deploy the function if it doesn't contain dependencies.